### PR TITLE
Add C# 9 tests for S3235, S4201, S2333, S3626, S1939, S3257 and S3440

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Related to #3668

### CODE REVIEW

**Comment by costin-zaharia-sonarsource** [comment]
I've changed this to draft until https://github.com/SonarSource/sonar-dotnet/pull/3712 is merged.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿int x = 5;
+int y = 6;
+
+y = y switch
+{
+    not 5 => 5 // Compliant - FN
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs:6]: could you also add a test with another switch arm?
how should it behave? raise only on the arm that is redundant, right?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantJumpStatement.CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+
+var a = new Action(() =>
```
Comment (andrei-epure-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantJumpStatement.CSharp9.cs:3]: is this related to C#  9 syntax? if not, maybe you can remove it

this functionality is tested at line 13

**Review by andrei-epure-sonarsource — APPROVED** [review]
LGTM - 2 comments

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantConditionalAroundAssignment.CSharp9.cs:6]: I'm not sure I understand how the other switch arm should look like

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantJumpStatement.CSharp9.cs (hunk 2 above)
Comment (costin-zaharia-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/RedundantJumpStatement.CSharp9.cs:3]: Indeed it is very similar with the one from line 13. I will remove it.
